tsconfig-type-test.json

```
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
+    "target": "es2018",
+    "module": "esnext",
```

I think this should be `nodenext` to handle the case of file extensions and all the other strict rules for Node.js ESM

```suggestion
    "module": "nodenext",
```